src/internal.c

```diff
@@ -6275,6 +6275,9 @@ void SSL_ResourceFree(WOLFSSL* ssl)
         ssl->dtls_rx_msg_list = NULL;
         ssl->dtls_rx_msg_list_sz = 0;
     }
+    XFREE(ssl->dtls_pending_finished, ssl->heap, DYNAMIC_TYPE_DTLS_BUFFER);
```

Comment:
Inconsistent heap hint type... Looks like this should be `DYNAMIC_TYPE_DTLS_MSG `. There are `DYNAMIC_TYPE_DTLS_MSG` and `DYNAMIC_TYPE_DTLS_BUFFER` both used for `dtls_pending_finished`.

Reply:
In the other free cases its wrapped with `if (ssl->dtls_pending_finished != NULL) {`. Perhaps you can make this one consistent?

Reply:
Done.